agent/discover/cluster.go

```diff
@@ -0,0 +1,91 @@
+package discover
+
+import (
+	"fmt"
+	"log"
+	"strconv"
+
+	// Reusing the Prometheus Ha Exporter cibadmin xml parser here
+	"github.com/ClusterLabs/ha_cluster_exporter/collector/pacemaker/cib"
+
+	consul "github.com/hashicorp/consul/api"
+	"github.com/pkg/errors"
+	consul_internal "github.com/trento-project/trento/internal/consul"
+)
+
+const crm_monPath string = "/usr/sbin/crm_mon"
+const cib_admPath string = "/usr/bin/cibadmin"
+
+//const crm_monPath string = "./test/fake/crm_mon.sh"
+//const cib_admPath string = "./test/fake_cibadmin.sh"
+
+// This Discover handles any Pacemaker Cluster type
+type ClusterDiscover struct {
+	cibConfig      cib.Root
+	clusterName    string
```

Comment:
As a comment, not all clusters have a name. I don't know if this impacts somehow the current implementation

Reply:
the clusterName can be empty as well. it simply is supposed to show what was configured. there is no reliance on the content being non-empty afaik.